yateto/codegen/common.py

```diff
@@ -122,3 +132,130 @@ def deduce_offset_arg(self, term):
       return '0'
     else:
       return f'{self.EXTRA_OFFSET_NAME}_{term.name}'
+
+class TinytcKernelArgument:
+
+  def __init__(self, name: str, call_expr: str, constant: bool, temporary: bool, modified: bool, offset: int = 0):
+    """Kernel argument for TinytcWrapper.
+
+    Arguments:
+    name -- Argument name
+    call_expr -- Expression used in calling wrapper
+    constant -- Whether a tensor is invariant to group id
+    temporary -- Whether a tensor is stored in a temporary buffer
+    modified -- Whether tensor is modified during kernel
+    """
+    self.name = name
```

Comment:
Just out of curiosity, why are there no underscores for class variables here but  used for TinytcWriter?

Reply:
I typically add underscores to members considered private as Python does not have any private members. Let me know if you find any inconsistencies in the naming.